src/common.h

```diff
@@ -113,6 +114,7 @@
 /* Max FLAC sample rate : https://xiph.org/flac/format.html */
 #define		SF_MAX_SAMPLERATE	655350
 
+#define		CLAMP_SHORT(x) (x > SHRT_MAX ? SHRT_MAX : x < SHRT_MIN ? SHRT_MIN : x)
```

Comment:
This isn't necessary when clamping float and it's potentially dangerous and inefficient (it can evaluate x up to 3 times).